### DISCUSSION TITLE
[FEAT] Merge Push, Email & SMS modules

### DISCUSSION THREAD

**Discussion by kkopanidis** [ConduitPlatform/Conduit#122] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently conduit spins up multiple containers that don't really make sense to be separate, since they can't practically scale independently. Email, push and sms utilise different external providers for their operations, but they fulfil a common goal of outgoing operations with templates etc.

**Describe the solution you'd like**
I think it'd be best for these to be merged into a singular "communications" module, that handles all outgoing user communications with either channel. This will allow for even cool things like "reach the user by any mean necessary", and incorporate more channels ex. viber, whatsapp etc.

**Additional context**
This particular module merge doesn't have any scalability arguments i think, since basically their capacity dependes 80-90% to the external services and merging them shouldn't provide any significant benefits or drawbacks on that part. But, it'd definitely save up some ram and cpu in the deployments.

**Comment by kon14** [comment]
> This will allow for even cool things like "reach the user by any mean necessary", and incorporate more channels ex. viber, whatsapp etc.

Nice one, definitely a feature we'd be interested in supporting.

This is also going to affect URIs much like #22.
